src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -1986,6 +1986,45 @@ protected override void AfterMembersCompletedChecks(BindingDiagnosticBag diagnos
                     var syntax = (ExtensionBlockDeclarationSyntax)this.GetNonNullSyntaxNode();
                     diagnostics.Add(ErrorCode.ERR_BadExtensionContainingType, syntax.Keyword);
                 }
+
+                if (TryGetOrCreateExtensionMarker() is { Parameters: [var extensionParameter] })
+                {
+                    checkUnderspecifiedGenericExtension(extensionParameter, TypeParameters, diagnostics);
```

Comment:
>checkUnderspecifiedGenericExtension [](http://example.com/codeflow?start=20&length=35)

I think that long term it would be better to perform this check per member and report an error per member. For example, once we enable indexers, I can imagine that some indexer declarations (referencing remaining type arguments in their parameters) will be allowed, and others won't be allowed.